src/format/parsed.rs

```diff
@@ -703,6 +647,74 @@ impl Parsed {
     }
 }
 
+fn resolve_week_number(
+    year: i32,
+    week: u32,
+    week_start_day: Weekday,
+    weekday: Weekday,
+) -> ParseResult<NaiveDate> {
+    let newyear = NaiveDate::from_yo_opt(year, 1).ok_or(OUT_OF_RANGE)?;
+    let firstweek =
+        (7 - newyear.weekday().num_days_from_sunday() + week_start_day.num_days_from_sunday()) % 7;
+    // `firstweek`-th day of January is the beginning of the week 1.
+
+    if week > 53 {
+        return Err(OUT_OF_RANGE);
+    }
+    let weekday_nr =
+        (weekday.num_days_from_sunday() + 7 - week_start_day.num_days_from_sunday()) % 7;
+    let ndays = firstweek as i32 + (week as i32 - 1) * 7 + weekday_nr as i32;
+    let date =
+        newyear.checked_add_signed(OldDuration::days(i64::from(ndays))).ok_or(OUT_OF_RANGE)?;
+    if date.year() != year {
+        return Err(OUT_OF_RANGE);
+    }
+    Ok(date)
+}
+
+fn resolve_year(y: Option<i32>, q: Option<i32>, r: Option<i32>) -> ParseResult<Option<i32>> {
```

Comment:
Please add a docstring for `resolve_year`.

Please use full variables names, e.g. `y` should be `year`, etc.
I don't know what `q` or `r` mean (I have some guesses, but it'd be better to use full words for those variables).

Tangential: a PR that type-aliased years, months, etc. would greatly increase code readability, i.e. `type year = i32;`, etc.

Reply:
> Tangential: a PR that type-aliased years, months, etc. would greatly increase code readability, i.e. `type year = i32;`, etc.

I quite like bare integers. But are you interested in trying out how it looks?